File: Documentation/4.0/Samples/java/src/test/java/net/ravendb/ClientApi/Operations/ClientConfig.java
package net.ravendb.ClientApi.Operations;

import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.operations.configuration.ClientConfiguration;
import net.ravendb.client.documents.operations.configuration.GetClientConfigurationOperation;
import net.ravendb.client.documents.operations.configuration.PutClientConfigurationOperation;
import net.ravendb.client.http.ReadBalanceBehavior;

public class ClientConfig {

    private interface IFoo {
        /*
        //region config_1_0
        GetClientConfigurationOperation()
        //endregion


        //region config_2_0
        PutClientConfigurationCommand(ClientConfiguration configuration)
        //endregion
        */
    }

    private static class Foo {
        //region config_1_1
        public static class Result {
            private long etag;
            private ClientConfiguration configuration;

            public long getEtag() {
                return etag;
            }

            public void setEtag(long etag) {
                this.etag = etag;
            }

            public ClientConfiguration getConfiguration() {
                return configuration;
            }

            public void setConfiguration(ClientConfiguration configuration) {
                this.configuration = configuration;
            }
        }
        //endregion
    }

    public ClientConfig() {
        try (IDocumentStore store = new DocumentStore()) {
            {
                //region config_1_2
                GetClientConfigurationOperation.Result config
                    = store.maintenance().send(new GetClientConfigurationOperation());
                ClientConfiguration clientConfiguration = config.getConfiguration();
                //endregion
            }

            {
                //region config_2_2
                ClientConfiguration clientConfiguration = new ClientConfiguration();
                clientConfiguration.setMaxNumberOfRequestsPerSession(100);
                clientConfiguration.setReadBalanceBehavior(ReadBalanceBehavior.FASTEST_NODE);

                store.maintenance().send(
                    new PutClientConfigurationOperation(clientConfiguration));
                //endregion
            }
        }
    }
}

File: Documentation/5.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/ClientConfig.cs
using System.Collections.Generic;
using Raven.Client.Documents;
using Raven.Client.Documents.Operations.Configuration;
using Raven.Client.Http;
using Raven.Client.ServerWide.Operations;
using Xunit;
using Raven.Client.ServerWide.Operations.Configuration;
using System.Net.Http;
using Raven.Client.Documents.Conventions;
using Raven.Client.Http;
using Raven.Client.Json;
using Raven.Client.Json.Serialization;
using Raven.Client.Util;
using Sparrow.Json;


namespace Raven.Documentation.Samples.ClientApi.Operations
{
    public class ClientConfig
    {
        private interface IFoo
        {
            /*
            #region config_1_0
            public GetClientConfigurationOperation()
            #endregion
            */

            /*
            #region config_2_0
            public PutClientConfigurationOperation(ClientConfiguration configuration)
            #endregion
            */
        }
        private class Foo
        {
            #region config_1_1
            public class Result
            {
                public long Etag { get; set; }

                public ClientConfiguration Configuration { get; set; }
            }
            #endregion

            #region DatabaseSettingsDefinition
            public class DatabaseSettings
            {
                public Dictionary<string, string> Settings { get; set; }
            }
            #endregion
        }



        public ClientConfig()
        {
            using (var store = new DocumentStore())
            {
                {
                    #region config_1_2
                    GetClientConfigurationOperation.Result config = store.Maintenance.Send(new GetClientConfigurationOperation());
                    ClientConfiguration clientConfiguration = config.Configuration;
                    #endregion
                }

                {
                    #region config_2_2
                    ClientConfiguration clientConfiguration = new ClientConfiguration
                    {
                        MaxNumberOfRequestsPerSession = 100,
                        ReadBalanceBehavior = ReadBalanceBehavior.FastestNode
                    };
                    
                    store.Maintenance.Send(new PutClientConfigurationOperation(clientConfiguration));
                    #endregion
                }


                {
                    #region ApplyDatabaseSettings-PutDatabaseSettingsOperation
                    // Configure database settings
                    static void PutConfigurationSettings(DocumentStore store, Dictionary<string, string> settings)
                    {
                        // Save the new settings with PutDatabaseSettingsOperation
                        store.Maintenance.Send(new PutDatabaseSettingsOperation(store.Database, settings));
                        // Disable the database
                        store.Maintenance.Server.Send(new ToggleDatabasesStateOperation(store.Database, true));
                        // Enable the database to apply new settings
                        store.Maintenance.Server.Send(new ToggleDatabasesStateOperation(store.Database, false));
                    }
                    #endregion
                }

                {
                    #region SeeNewDatabaseSettings-GetDatabaseSettingsOperation
                    static DatabaseSettings GetConfigurationSettings(DocumentStore store)
                    {
                        var settings = store.Maintenance.Send(new GetDatabaseSettingsOperation(store.Database));
                        Assert.NotNull(settings);
                        return settings;
                    }
                    #endregion
                }

                {
                    static void
                    #region signature-PutDatabaseSettingsOperation
                    PutDatabaseSettingsOperation(string databaseName, Dictionary<string, string> configurationSettings)
                    #endregion
                    {

                    }

                    static void
                    #region signature-ToggleDatabasesStateOperation
                    ToggleDatabasesStateOperation(string databaseName, bool disable)
                    #endregion
                    {

                    }

                    static void
                    #region signature-GetDatabaseSettingsOperation
                    GetDatabaseSettingsOperation(string databaseName)
                    #endregion
                    {

                    }
                }

            }
        }
    }
}
